CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.12.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.17.0 FATAL_ERROR)
```

Comment:
Are you sure?

Reply:
@zeule Only needed for the changes introduced by the second commit. See https://cmake.org/cmake/help/v3.17/release/3.17.html#modules (the part about `FindPython3`).

If it were possible to set the minimum required version just for the python part of the build that would be ideal, but AFAIK that's not possible, so the version requirement has to be bumped for the whole project.

Reply:
You can set `cmake_minimum_required(VERSION 3.17.0)` in `bindings/python/CMakeLists.txt` but 3.12->3.17 is still a huge jump, that might be better for master branch instead. People shouldn't _need_ to be bleeding edge to build libtorrent.

Reply:
@FranciscoPombal  uses `WITH_SOABI` option to `python_add_module()`, which appeared in 3.17. Without the option this PR makes little sense.

Reply:
Which is also used within `bindings/python/CMakeLists.txt`, so you can still get away with setting the minimum in that file versus bumping it globally.

Reply:
@NotTsunami

> You can set `cmake_minimum_required(VERSION 3.17.0)` in `bindings/python/CMakeLists.txt`

You're right, it works after all. Fixed.

> but 3.12->3.17 is still a huge jump, that might be better for master branch instead. People shouldn't need to be bleeding edge to build libtorrent.

the version bump is required for the changes in the second commit, and only when building python bindings. As for the bleeding edge argument, I don't think it is a convincing one. I think there is no good reason for anyone _not_ to use the latest CMake version. See a previous comment I made about that: https://github.com/qbittorrent/qBittorrent/pull/12480#issuecomment-612514368, which explains my reasoning in detail.